lib/rubygems/test_utilities.rb

```diff
@@ -48,7 +48,14 @@ def find_data(path)
       raise Gem::RemoteFetcher::FetchError.new("no data for #{path}", path)
     end
 
-    @data[path]
+    data = @data[path]
+
+    if data.respond_to?(:flatten!)
+      data.flatten!
+      data.size <= nargs ? data.last(nargs) : data.shift(nargs)
```

Comment:
I mean the last 3 values should be kept regardless of how many times they are fetched. Otherwise the fake response array would be empty and cause error.

Reply:
> the last 3 values should be kept regardless of how many times they are fetched

is this a requirement for tests you have added (since `flatten` would be true only for nested array) ? if yes, explicitly declare it in test. ie
```
ok_res = [response_success, 200, 'OK']
@stub_fetcher.data["..."] = [
     [response_fail, 401, 'Unauthorized'],
     ok_res,
     ok_res,
     ...
    ]
```

IMHO, when testing, some repetition is tolerable if it makes things simpler to understand.